src/OpenTelemetry.AutoInstrumentation/Configurations/NoCodeSettings.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration;
+using OpenTelemetry.AutoInstrumentation.Logging;
+using static OpenTelemetry.AutoInstrumentation.InstrumentationDefinitions;
+
+namespace OpenTelemetry.AutoInstrumentation.Configurations;
+
+internal class NoCodeSettings : Settings
+{
+    private static readonly IOtelLogger Log = OtelLogging.GetLogger();
+
+    public bool Enabled { get; set; }
+
+    public List<NoCodeInstrumentedMethod> InstrumentedMethods { get; set; } = [];
+
+    public Payload GetDirectPayload()
+    {
+        return new Payload
+        {
+            // Fixed Id for definitions payload (to avoid loading same integrations from multiple AppDomains)
+            DefinitionsId = "D3B88A224E034D60AC3A923BABEE6B7F",
+            Definitions = InstrumentedMethods.Select(x => x.Definition).ToArray(),
+        };
+    }
+
+    protected override void OnLoadEnvVar(Configuration configuration)
+    {
+        // Intentionally empty. NoCodeSettings are only loaded from file-based configuration.
+    }
+
+    protected override void OnLoadFile(YamlConfiguration configuration)
+    {
+        var noCodeConfiguration = configuration.NoCode;
+
+        if (noCodeConfiguration is null)
+        {
+            Log.Debug("NoCode configuration section not found, no-code instrumentation will be disabled.");
+            return;
+        }
+
+        if (noCodeConfiguration.Targets == null || noCodeConfiguration.Targets.Length == 0)
+        {
+            Log.Debug("NoCode targets section no found, or it is empty, no-code instrumentation will be disabled.");
+            return;
+        }
+
+        var noCodeEntries = noCodeConfiguration.Targets;
+
+        var instrumentedMethods = new List<NoCodeInstrumentedMethod>(noCodeEntries.Length);
+
+        foreach (var noCodeEntry in noCodeEntries)
+        {
+            var noCodeTarget = noCodeEntry.Target;
+
+            if (noCodeTarget == null)
+            {
+                Log.Debug("No code target option is null. Skipping this entry.");
```

Comment:
Shouldn't this be warning? `this entry` would be unknown marker to find the broken line.